tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/AsynchronousTest.java

```diff
@@ -65,34 +70,21 @@ public static WebArchive deploy() {
      */
     @Test
     public void testAsyncIsNotFinished() {
-        Future<Connection> future = null;
-        try {
-            future = client.service();
-        }
-        catch (InterruptedException e) {
-            throw new AssertionError("testAsync: unexpected InterruptedException calling service");
-        }
-
-        Assert.assertFalse(future.isDone());
-
+        CompletableFuture<Void> waitingFuture = newWaitingFuture();
+        CompletionStage<Connection> resultFuture = client.serviceCS(waitingFuture);
+        Future<Connection> future = resultFuture.toCompletableFuture();
+        await().atMost(200, TimeUnit.MILLISECONDS).untilAsserted(()-> Assert.assertFalse(future.isDone()));
     }
 
     /**
      * Test that the future returned by calling an asynchronous method is done if called after waiting enough time to end the operation
      */
     @Test
     public void testAsyncIsFinished() {
-        Future<Connection> future = null;
-        try {
-            future = client.service();
-            Thread.sleep(1500);
-        }
-        catch (InterruptedException e) {
-            throw new AssertionError("testAsync: unexpected InterruptedException calling service");
-        }
-
-        Assert.assertTrue(future.isDone());
-
+        CompletableFuture<Void> waitingFuture = newWaitingFuture();
+        CompletionStage<Connection> resultFuture = client.serviceCS(waitingFuture);
+        Future<Connection> future = resultFuture.toCompletableFuture();
+        await().atLeast(1000, TimeUnit.MILLISECONDS).untilAsserted(()-> Assert.assertTrue(future.isDone()));
```

Comment:
Assuming there is a `service` method taking `waitingFuture` and returning `Future`, I think this test should look roughly like this:

```java
CompletableFuture<Void> waitingFuture = newWaitingFuture();
Future<Connection> future = client.service(waitingFuture);
assertFalse(future.isDone());
waitingFuture.complete(null);
await().atMost(1000, TimeUnit.MILLISECONDS).untilAsserted(()-> Assert.assertTrue(future.isDone()));
```

Reply:
Alternatively, if we don't want to complete the wait condition and we're fine with useless 1 second wait inside the service method, this would be equivalent I think:

```java
CompletableFuture<Void> waitingFuture = newWaitingFuture();
Future<Connection> future = client.service(waitingFuture);
assertFalse(future.isDone());
await().atLeast(1000, TimeUnit.MILLISECONDS).atMost(2000, TimeUnit.MILLISECONDS).untilAsserted(()-> Assert.assertTrue(future.isDone()));
```

Reply:
And the `assertFalse(future.isDone());` is not 100% necessary, that's covered by the other test. But it shouldn't hurt :-)

Reply:
done